### DISCUSSION TITLE
DashBoard Disappears. Only Heading remians

### DISCUSSION THREAD

**Discussion by zbadshah** [filamentphp/filament#5571] [Help]
Apparently, I did Composer update and all DashBoard menu disappeared, i.e. no resources are showing except the headings.

What could have caused this?

Will highly appreciate help as I spent a lot of time looking for issues and could not find.

**Comment by danharrin** [comment]
Clear your view cache and delete any published views in `resources/views/vendor/filament`. Then clear your browser's local storage.

**Comment by zbadshah** [comment]
Thanks Dan. I deleted the bootstrap\cache and recreated it. Deleted all published views. Also cleared browser cache. No change. Then I used another browser and still no change. 

I do hope that I did all steps as per your suggestion, but it appears that there is more to it.

**Reply by danharrin** [reply]
Try reproducing it in a new Laravel/Filament app on a different domain. Its definitely a cache issue.

**Reply by zbadshah** [reply]
> Try reproducing it in a new Laravel/Filament app on a different domain. Its definitely a cache issue.

Kindly help to solve $this->halt giving error after moving to new domain. It does not work for any life-cycle hook and mutate functions.

**Reply by zbadshah** [reply]
@danharrin 
Hi Dan, Still facing $this->halt issue on a new installation as was suggested by you earlier. The disappearance of the menu was solved but this problem came up.

The error is as follow:
Property [$halt] not found on component: [app.filament.resources.project-milestone-resource.pages.create-project-milestone]

The installation is of course updated so need your help to resolve. Thanks

**Comment by sergioraldana** [comment]
Did you install plugins? Did you try to switch dark mode already ?

**Reply by zbadshah** [reply]
I created another installation and copied the resources. It worked. But $this->halt is not working on life cycle hooks as well as mutate.

Kindly let me know. I did filament upgrade, composer update and config:cache etc. It did not work.

**Reply by zbadshah** [reply]
> Did you install plugins? Did you try to switch dark mode already ?

I am using spatie permissions and medialibrary. I am also using dark mode,

**Comment by discoverlance-com** [comment]
Any solution to this issue? I am also having the same one

**Reply by zbadshah** [reply]
> Any solution to this issue? I am also having the same one

which issue? disappearance of dashboard or $this->halt not working?

**Reply by discoverlance-com** [reply]
> > Any solution to this issue? I am also having the same one
> 
> which issue? disappearance of dashboard or $this->halt not working?

the disappearance of the dashboard

**Reply by zbadshah** [reply]
> > > Any solution to this issue? I am also having the same one
> > 
> > 
> > which issue? disappearance of dashboard or $this->halt not working?
> 
> the disappearance of the dashboard

The first solution suggested above by Dan did not work. So he suggested to create a fresh installation and copy resources there. It worked for me but $this->halt function is not working. I have asked above but there has been no response.

**Reply by discoverlance-com** [reply]
I am now able to get my app working. I cleared out my published filament files, did a diff to the old one and realized there were some differences between them. So by updating to the new ones, I resolved the issue.

**Comment by jwhoami** [comment]
I am not sure if this could be the case or someone runs into a case similar to mine, but something similar happened to me where the dashboard disappeared.

I had imported alpinejs in my app.js but since aplinejs is already imported by filament, it was creating some kind of
inconsistencies resulting in this weird behavior.

Removing alpinejs from my app.js resolved the issue in my case.

**Reply by zbadshah** [reply]
> I am not sure if this could be the case or someone runs into a case similar to mine, but something similar happened to me where the dashboard disappeared.
> 
> I had imported alpinejs in my app.js but since aplinejs is already imported by filament, it was creating some kind of inconsistencies resulting in this weird behavior.
> 
> Removing alpinejs from my app.js resolved the issue in my case.

I am not using anything new. Just by composer update or clearing cache this happened. I still could not solve it. However, creating a new instance solved the problem. It is essential to find the solution as it is very annoying and time wasting. Will update if I found it.

**Reply by discoverlance-com** [reply]
I found out recently that the views get updated sometimes when filament releases new updates so any published views might be outdated. So for now best way is delete all filament views and republish. If you have vscode for instance, you can use the git tool in your toolbar pane(on right or left) to look at the new changes to the views. And see if they conflict with any of your previous changes and make relevant changes.

At the moment I think it's difficult to do otherwise because if new updates require changes in views the contributors and maintainers cannot say that because users are already using or published the views they will not update it when it's required for an update. And if vendor publish command for views overwrites the currently published views automatically, it might not be desired as users might have changes they want to maintain 

The other way to ensure you don't have this problem would probably be not to update filament if you don't necessarily have any bug or security issue that has been fixed.

**Reply by danharrin** [reply]
Please don't publish views if you don't need to... and if you do, delete any views you don't publish

**Reply by danharrin** [reply]
https://filamentphp.com/blog/publishing-views-in-laravel
